### ISSUE TITLE
Rendering bug for PostgreSQL

### ISSUE THREAD

**Issue by mjf** [okbob/pspg#162]
`\d table`

![bug](https://user-images.githubusercontent.com/54104/114356060-76667280-9b70-11eb-8508-1f61cf8ff8ad.png)

The last line does not get render properly.

```
$ pspg --version
pspg-4.5.0
with readline (version: 0x0700)
with integrated menu
ncurses version: 6.1, patch: 20180224
ncurses with wide char support
ncurses widechar num: 1
wchar_t width: 4, max: 2147483647
with postgres client integration
with inotify support
```

**Comment by okbob** [comment]
po 12. 4. 2021 v 9:23 odesílatel Matouš Jan Fialka ***@***.***>
napsal:

> \d table
>
> [image: bug]
> <https://user-images.githubusercontent.com/54104/114356060-76667280-9b70-11eb-8508-1f61cf8ff8ad.png>
>
> The last line does not get render properly.
>

I can reproduce this issue. I'll look if I can fix it. In this case is hard
to detect end of table and start of status line.

There is simple workaround - use border 2 instead 1. Then pspg can detect
end of table

Thank you for bug report

Pavel

**Comment by mjf** [comment]
> There is simple workaround - use border 2 instead 1. Then pspg can detect
  end of table

```sh
$ grep border .pspgconf 
force_uniborder = true
border_type = 2
```

**Comment by okbob** [comment]
po 12. 4. 2021 v 14:14 odesílatel Matouš Jan Fialka <
***@***.***> napsal:

> There is simple workaround - use border 2 instead 1. Then pspg can detect
> end of table
>
> $ grep border .pspgconf
> force_uniborder = true
> border_type = 2
>
> you need to change it in psql side

set ~/.psqlrc

\pset border 2

the change of border is too significant change of output format - and it
cannot be done in pspg.

**Comment by mjf** [comment]
I also realized meanwhile what you meant with the border to be set to 2. Beg my pardon then, please. :smile:

An idea... it seems to me that there is always either `/^[^:]+:$/` or `/^$/` (empty line) after a table data output...

**Comment by mjf** [comment]
> There is simple workaround - use border 2 instead 1.

This works as charm, of course, but does not look so pretty to me I must say.

**Comment by okbob** [comment]
po 12. 4. 2021 v 14:42 odesílatel Matouš Jan Fialka <
***@***.***> napsal:

> I also realized meanwhile what you meant with the border to be set to 2.
> Beg my pardon then, please. 😄
>
> An idea... it seems to me that there is always either /^[^:]+:$/ or /^$/
> (empty line) after a table data output...
>
Empty line is after the complete result. But tabular data are not separated


                  Table "public.xx"
 Column │  Type   │ Collation │ Nullable │ Default
════════╪═════════╪═══════════╪══════════╪═════════
 a      │ integer │           │ not null │
Indexes:
    "xx_pkey" PRIMARY KEY, btree (a)

**Comment by okbob** [comment]
po 12. 4. 2021 v 14:44 odesílatel Matouš Jan Fialka <
***@***.***> napsal:

> There is simple workaround - use border 2 instead 1.
>
> This works as charm, of course, but does not look so pretty to me I must
> say.
>
sure - visual preferences are subjective :) and it is the reason why I
wrote themes there.

**Comment by okbob** [comment]
po 12. 4. 2021 v 14:45 odesílatel Pavel Stehule ***@***.***>
napsal:

>
>
> po 12. 4. 2021 v 14:42 odesílatel Matouš Jan Fialka <
> ***@***.***> napsal:
>
>> I also realized meanwhile what you meant with the border to be set to 2.
>> Beg my pardon then, please. 😄
>>
>> An idea... it seems to me that there is always either /^[^:]+:$/ or /^$/
>> (empty line) after a table data output...
>>
> Empty line is after the complete result. But tabular data are not
> separated
>
>
>                   Table "public.xx"
>  Column │  Type   │ Collation │ Nullable │ Default
> ════════╪═════════╪═══════════╪══════════╪═════════
>  a      │ integer │           │ not null │
> Indexes:
>     "xx_pkey" PRIMARY KEY, btree (a)
>

data rows in border = 1 mode started with space, so the fix should not be
hard.

Pavel

**Comment by okbob** [comment]
should be fixed by 82d28c0c5063078cc83ddeacc01c5d0514580ff7 commit - please, check

**Comment by okbob** [comment]
fixed in 4.6.0
